official/wide_deep/movie_dataset.py

```diff
@@ -0,0 +1,303 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Download Kaggle movie dataset and construct input_fn."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import ast
+import hashlib
+import os
+import subprocess
+import sys
+import tempfile
+import zipfile
+
+# pylint: disable=wrong-import-order
+from absl import app as absl_app
+from absl import flags
+import numpy as np
+import pandas as pd
+import six
+from six.moves import urllib  # pylint: disable=redefined-builtin
+import tensorflow as tf
+# pylint: enable=wrong-import-order
+
+from official.utils.flags import core as flags_core
+
+
+_RATINGS = "ratings.csv"
+_RATINGS_SMALL = "ratings_small.csv"
+_METADATA = "movies_metadata.csv"
+_FILES = ["credits.csv", "keywords.csv", "links.csv", "links_small.csv",
+          _METADATA, "ratings.csv", "ratings_small.csv"]
+_ZIP = "the-movies-dataset.zip"
+
+
+_BUFFER_SUBDIR = "data_buffer"
+_FEATURE_MAP = {
+    "userId": tf.FixedLenFeature([1], dtype=tf.int64),
+    "movieId": tf.FixedLenFeature([1], dtype=tf.int64),
+    "year": tf.FixedLenFeature([1], dtype=tf.int64),
+    "original_language": tf.FixedLenFeature([], dtype=tf.string),
+    "genres_0": tf.FixedLenFeature([], dtype=tf.string),
+    "genres_1": tf.FixedLenFeature([], dtype=tf.string),
+    "budget": tf.FixedLenFeature([1], dtype=tf.float32),
+    "rating": tf.FixedLenFeature([1], dtype=tf.float32)
+}
+
+_BUFFER_SIZE = {
```

Comment:
What is this buffer size/subdir used for?

Reply:
It is used to validate the binary representation of the data. So if the file construction is interrupted midway through, it won't have the same number of bytes and the program will know to reconstruct it.